corellium/client/src/main/kotlin/flank/corellium/client/agent/Connect.kt

```diff
@@ -0,0 +1,99 @@
+package flank.corellium.client.agent
+
+import flank.corellium.client.data.AgentOperation
+import flank.corellium.client.data.CommandResult
+import flank.corellium.client.util.withProgress
+import io.ktor.client.HttpClient
+import io.ktor.client.features.logging.LogLevel
+import io.ktor.client.features.logging.Logging
+import io.ktor.client.features.websocket.DefaultClientWebSocketSession
+import io.ktor.client.features.websocket.WebSockets
+import io.ktor.client.features.websocket.webSocketSession
+import io.ktor.client.request.header
+import io.ktor.client.request.url
+import io.ktor.http.cio.websocket.Frame
+import io.ktor.http.cio.websocket.readText
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withTimeout
+import kotlinx.serialization.decodeFromString
+
+suspend fun connectAgent(
+    agentUrl: String,
+    token: String,
+    logLevel: LogLevel = LogLevel.NONE
+): Agent = withProgress {
+    val client = HttpClient {
+        install(WebSockets)
+        install(Logging) {
+            level = logLevel
+        }
+    }
+
+    var connection: Agent? = null
+    do {
+        try {
+            connection = client
+                .createSession(agentUrl, token)
+                .let(::Agent)
+                .apply {
+                    handleIncomingFrames()
+                    waitForReady()
+                }
+        } catch (ex: Exception) {
+            ex.printStackTrace()
+            delay(20_000)
+        }
+    } while (connection == null)
```

Comment:
possible infinite loop?

Reply:
fixed by adding require at the beginning of `do`